src/filerepository.js

```diff
@@ -0,0 +1,455 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module upload/filerepository
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+
+import CKEditorError from '@ckeditor/ckeditor5-utils/src/ckeditorerror';
+import ObservableMixin from '@ckeditor/ckeditor5-utils/src/observablemixin';
+import Collection from '@ckeditor/ckeditor5-utils/src/collection';
+import mix from '@ckeditor/ckeditor5-utils/src/mix';
+import log from '@ckeditor/ckeditor5-utils/src/log';
+
+import FileReader from './filereader.js';
+
+import uid from '@ckeditor/ckeditor5-utils/src/uid.js';
+
+/**
+ * FileRepository plugin.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class FileRepository extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get pluginName() {
+		return 'fileRepository';
```

Comment:
> according to the recent convention should it be "upload/fileRepository"? I

That's not the recent convention. That's the convention we had from the beginning. And it should be the module name – `upload/filerepository`.



> They make names much longer

Really? :D I counted 7 characters here.

> does not let overwrite plugins nicely in the future and might be a problem if we will try to move plugins between packages (note that from the end-developer perspective it will be the only thing which will change).

If we'll move plugins between repositories we'll have a lot more issues.

We scope packages for the same reason why we scope packages – to avoid accidental conflicts (although, in case of plugins we're scoping by the pkg name, not the org name, because, if we added org name, then, indeed, the name would be awfully long). Your plugin may work with `x/y` but it may not work with `x2/y` if one will load it to his/her editor. So if you retrieved the plugin by `y` you'd blow up the whole thing.

If, for some reason, someone will ever decide to write a copy of plugin `x/y` which is supposed to replace it, then he/she can name it like this. But this should be intentional, not accidental.

PS. Strategy for naming contributing plugins is yet to be defined.